### PR TITLE
Add a module ExchangeNavigationHelper to the test framework to factorize navigation

### PR DESCRIPTION
**Pull request by fbeutin-ledger**

### CODE REVIEW

File: test/python/apps/exchange_navigation_helper.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
+from pathlib import Path
+
+from ragger.navigator import Navigator, NavInsID
+from ragger.backend import BackendInterface
+
+ROOT_SCREENSHOT_PATH = Path(__file__).parent.parent.resolve()
+
+class ExchangeNavigationHelper:
```
Comment (lpascal-ledger) [inline test/python/apps/exchange_navigation_helper.py:8]: The `Navigator` is also an helper firstly (as most if not all abstraction code in general). Naming this class `ExchangeNavigator` feels enough for me. I'm no big fan of `LongNameWithoutMuchValueAddedInIt`

File: test/python/test_fund_flow_ethereum.py [hunk 2]
```diff
@@ -1,14 +1,10 @@
...
 from .apps.exchange import ExchangeClient, Rate, SubCommand
 from .apps.ethereum import EthereumClient
 
 from .signing_authority import SigningAuthority, LEDGER_SIGNER
 
-from .utils import ROOT_SCREENSHOT_PATH
-
 
-def test_fund_flow_ethereum_max_partner_name_length(backend, firmware, navigator, test_name):
+def test_fund_flow_ethereum_max_partner_name_length(backend, exchange_navigation_helper):
```
Comment (lpascal-ledger) [inline test/python/test_fund_flow_ethereum.py:7]: `exchange_navigator` feels good enough. Of course its a helper.

File: test/python/apps/exchange_navigation_helper.py
Comment (lpascal-ledger) [inline test/python/apps/exchange_navigation_helper.py:1]: `exchange_navigator.py`? Sorry for the ranting. But I don't get what this `_helper` postfix brings, except clustering the horizontal line.
If the object instantiated from the class defined here is the go-to instance to manage all the navigation of exchange, then its the exchange navigator itself.

**Review by lpascal-ledger — APPROVED** [review]
All good - except, in my opinion, some superfluously long names.

File: test/python/apps/exchange_navigation_helper.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        snapshots_dir_name = self._test_name
+        screen_change_after_last_instruction = True
+
+        if self._backend.firmware.device.startswith("nano"):
+            navigate_instruction = NavInsID.RIGHT_CLICK
+            validation_instructions = [NavInsID.BOTH_CLICK]
+            if accept:
+                text = "Accept"
+            else:
+                text = "Reject"
```
[the remark this answers is not in the document]
Comment (fbeutin-ledger) [reply test/python/apps/exchange_navigation_helper.py:25]: Done

File: test/python/apps/exchange_navigation_helper.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        self._backend = backend
+        self._navigator = navigator
+        self._test_name = test_name
+
+    def _navigate_and_compare(self, accept: bool):
+        # Default behaviors
+        snapshots_dir_name = self._test_name
+        screen_change_after_last_instruction = True
+
+        if self._backend.firmware.device.startswith("nano"):
```
[the remark this answers is not in the document]
Comment (fbeutin-ledger) [reply test/python/apps/exchange_navigation_helper.py:19]: Done

**Pull request merged**